### PR TITLE
feat(backend): make SPSP optional

### LINKED ISSUE #607: Make SPSP-receiving support optional
Add a backend config flag to enable/disable receiving via SPSP to Open Payments payment pointers.

### PR DESCRIPTION
**Pull request by sabineschaller**
## Changes proposed in this pull request

- adds environment variable to enable/disable SPSP

## Context

fixes #607 

## Checklist

- [x] Related issues linked using `fixes #number`
- [x] Tests added/updated
- [x] Documentation added
- [ ] Make sure that all checks pass
- [ ] Bruno collection updated

### CODE REVIEW

File: packages/backend/src/payment-method/ilp/spsp/middleware.ts [hunk 1]
```diff
@@ -5,7 +5,20 @@ export type SPSPWalletAddressContext = WalletAddressContext &
     incomingPayment?: never
   }
 
-export const spspMiddleware = async (
+export function createSpspMiddleware(spspEnabled: boolean) {
```
Comment (mkurapov) [inline packages/backend/src/payment-method/ilp/spsp/middleware.ts:8]: :+1:

File: packages/backend/src/payment-method/ilp/spsp/middleware.test.ts [hunk 2]
```diff
@@ -21,6 +25,7 @@ describe('SPSP Middleware', (): void => {
 
   beforeEach((): void => {
     next = jest.fn()
+    spspMiddleware = createSpspMiddleware(true)
```
Comment (mkurapov) [inline packages/backend/src/payment-method/ilp/spsp/middleware.test.ts:28]: can we test the `false` path here?

File: packages/documentation/src/content/docs/integration/deployment.md [hunk 3]
```diff
@@ -61,6 +61,7 @@ Now, the Admin UI can be found on localhost:3010.
...
 | `ENABLE_AUTO_PEERING`                                 |                                                                                                                                         | `false`                                                     | Flag to enable auto peering. View [documentation](/concepts/interledger-protocol/peering#auto-peering).                                                                                                    |
+| `ENABLE_SPSP`                                         |                                                                                                                                         | `true`                                                      | enables SPSP route                                                                                                                                                                                         |
```
Comment (mkurapov) [inline packages/documentation/src/content/docs/integration/deployment.md:64]: should we link to SPSP glossary item here?

File: packages/backend/src/payment-method/ilp/spsp/middleware.test.ts (hunk 2 above)
Comment (sabineschaller) [reply packages/backend/src/payment-method/ilp/spsp/middleware.test.ts:28]: You mean that next is called without anything happening?

File: packages/documentation/src/content/docs/integration/deployment.md (hunk 3 above)
Comment (sabineschaller) [reply packages/documentation/src/content/docs/integration/deployment.md:64]: Sure. I didn't really know what to put there. Was hoping the writers had a good idea ;-)

File: packages/backend/src/payment-method/ilp/spsp/middleware.test.ts (hunk 2 above)
Comment (mkurapov) [reply packages/backend/src/payment-method/ilp/spsp/middleware.test.ts:28]: Yes, a boring but good test to have

**Review by BlairCurrey — APPROVED** [review]
Looks good to me although I figure we are waiting to update the ENV var description. 

Total nit but I think changing `spspMiddleware`'s condition to be this would be logically equivalent and maybe simpler? Maybe not even worth changing what you have though TBH.

```ts
if (ctx.accepts('application/spsp4+json') && spspEnabled) {
```

new createSpspMiddleware fn:

| spspEnabled | accepts application/spsps4+json | result       |
|-------------|---------------------------------|--------------|
|      1      |                 1               | spsps get called |
|      1      |                 0               | next called  |
|      0      |                 1               | next called  |
|      0      |                 0               | next called  |

adding condition to spspMiddleware:

| spspEnabled | accepts application/spsps4+json | result       |
|-------------|---------------------------------|--------------|
|      1      |                 1               | spsps get called |
|      1      |                 0               | next called  |
|      0      |                 1               | next called  |
|      0      |                 0               | next called  |

**Comment by sabineschaller** [comment]
>Total nit but I think changing spspMiddleware's condition to be this would be logically equivalent and maybe simpler? Or maybe this is just how my brain works and not even worth changing what you have TBH.

I'll add it to the function, since I can't pass the config variable to the middleware.

**Review by mkurapov — APPROVED** [review]

**Pull request merged**